demo_example/async_echo_client.cpp

```diff
@@ -29,28 +70,14 @@ async_simple::coro::Lazy<void> start(asio::io_context &io_context,
         throw asio::system_error(ec);
     }
     std::cout << "Connect to " << host << ":" << port << " successfully.\n";
-    const int max_length = 1024;
-    char write_buf[max_length] = {"hello async_simple"};
-    char read_buf[max_length];
-    const int count = 100000;
-    for (int i = 0; i < count; ++i) {
-        co_await async_write(socket, asio::buffer(write_buf, max_length));
-        auto [error, reply_length] = co_await async_read_some(
-            socket, asio::buffer(read_buf, max_length));
-        if (error == asio::error::eof) {
-            std::cout << "eof at message index: " << i << '\n';
-            break;
-        } else if (error) {
-            std::cout << "error: " << error.message()
-                      << ", message index: " << i << '\n';
-            throw asio::system_error(error);
-        }
 
-        // handle read data as your wish.
+    std::vector<async_simple::coro::Lazy<>> input;
+    for (int i = 0; i < concurrency; ++i) {
+        input.push_back(task(socket));
     }
+    input.push_back(show_qps());
+    co_await async_simple::coro::collectAll(std::move(input));
```

Comment:
collectAll run multi coroutines in same thread asynchronously, looks like you want all coroutines parallel on all thread, try to use `collectAllPara`？